src/vdom/diff.js

```diff
@@ -187,10 +187,10 @@ function innerDiffNode(dom, vchildren, context, mountAll, absorb) {
 		min = 0,
 		len = originalChildren.length,
 		childrenLen = 0,
-		vlen = vchildren && vchildren.length,
+		vlen = vchildren ? vchildren.length : 0,
 		j, c, vchild, child;
 
-	if (len) {
+	if (len!==0) {
```

Comment:
While you're making things more explicit, don't you really mean `len > 0` in both cases?